cmd/apiserver/app/server/util.go

```diff
@@ -17,46 +17,165 @@ limitations under the License.
 package server
 
 import (
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
 	"github.com/golang/glog"
 
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/admission"
 	genericapiserver "k8s.io/apiserver/pkg/server"
+	kubeinformers "k8s.io/client-go/informers"
+	kubeclientset "k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+
+	scadmission "github.com/kubernetes-incubator/service-catalog/pkg/apiserver/admission"
+	"github.com/kubernetes-incubator/service-catalog/pkg/client/clientset_generated/internalclientset"
+	informers "github.com/kubernetes-incubator/service-catalog/pkg/client/informers_generated/internalversion"
+	"github.com/kubernetes-incubator/service-catalog/pkg/version"
 )
 
-func setupBasicServer(s *ServiceCatalogServerOptions) (*genericapiserver.Config, error) {
+// serviceCatalogConfig is a placeholder for configuration
+type serviceCatalogConfig struct {
+	// the shared informers that know how to speak back to this apiserver
+	sharedInformers informers.SharedInformerFactory
+	// the shared informers that know how to speak back to kube apiserver
+	kubeSharedInformers kubeinformers.SharedInformerFactory
+	// the configured loopback client for this apiserver
+	client internalclientset.Interface
+	// the configured client for kube apiserver
+	kubeClient kubeclientset.Interface
+}
+
+// standaloneMode returns true if the env var SERVICE_CATALOG_STANDALONE=true
+// if enabled, we will assume no integration with kubernetes
+// api server is available.  this is primarily useful for testing purposes.
+func standaloneMode() bool {
+	val := os.Getenv("SERVICE_CATALOG_STANDALONE")
+	return val == "true"
+}
```

Comment:
can you put this into `(./cmd/apiserver/app/server).ServiceCatalogServerOptions` along with all the rest of the options?